plugins/trace_api_plugin/abi_data_handler.cpp

```diff
@@ -0,0 +1,27 @@
+#include <eosio/trace_api_plugin/abi_data_handler.hpp>
+#include <eosio/chain/abi_serializer.hpp>
+#include <eosio/trace_api_plugin/request_handler.hpp>
+
+namespace eosio::trace_api_plugin {
+
+   void abi_data_handler::add_abi( const chain::name& name, const chain::abi_def& abi ) {
+      abi_serializer_by_account.emplace(name, std::make_shared<chain::abi_serializer>(abi, fc::microseconds::maximum()));
+   }
+
+   fc::variant abi_data_handler::process_data(const action_trace_v0& action, const fc::time_point& deadline) {
+      if (abi_serializer_by_account.count(action.account) > 0) {
+         const auto& serializer_p = abi_serializer_by_account.at(action.account);
+         auto type_name = serializer_p->get_action_type(action.action);
+
+         if (!type_name.empty()) {
+            try {
+               return serializer_p->binary_to_variant(type_name, action.data, fc::microseconds::maximum());
```

Comment:
Why are you not using `deadline` here?

Reply:
adding resolution from the live discussion here:

Since the operators will directly control available ABIs for the first deliverable, we can leave this as `fc::microseconds::maximum()` .  In the future version where ABIs are not implicitly trusted (eg they are derived from the running chain) we will need to enforce a deadline. 

At that time, we will likely make intrusive changes to `abi_serializer` to support the concept of a caller defined `yield` function replacing the internally plumbed time checks.  This allows separation of concerns over why a stalled/long-running call needs to terminate from the actual logic of `abi_serializer`.  This does a few important things:
* it allows the concept of "time" to be mocked in tests for determinism instead of being deeply coupled to the system clock etc
* it allows interruption for reasons other than time, for instance, a critical priority task needs to interrupt ABI processing occasionally and restricting _all_ ABI processing to short time slices to service that response time guarantee is too aggressive.